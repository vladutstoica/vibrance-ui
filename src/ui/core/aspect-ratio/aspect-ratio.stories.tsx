import type { Meta, StoryObj } from "@storybook/react";
import * as React from "react";
import { AspectRatio } from "./aspect-ratio";

const meta: Meta<typeof AspectRatio> = {
  title: "Design System/Core/AspectRatio",
  component: AspectRatio,
  tags: ["autodocs"],
  argTypes: {
    ratio: {
      description: "The aspect ratio of the container",
      control: "text",
      table: {
        type: { summary: "string" },
        defaultValue: { summary: "1/1" },
      },
    },
  },
};

export default meta;
type Story = StoryObj<typeof AspectRatio>;

export const Square: Story = {
  render: () => (
    <div className="w-[500px]">
      <AspectRatio ratio={1} className="bg-muted">
        <img
          src="https://placehold.co/500"
          alt="Placeholder Alt Text"
          className="rounded-md object-cover"
        />
      </AspectRatio>
    </div>
  ),
};

export const Landscape: Story = {
  render: () => (
    <div className="w-[500px]">
      <AspectRatio ratio={16 / 9} className="bg-muted">
        <img
          src="https://placehold.co/1600x900"
          alt="Placeholder Alt Text"
          className="rounded-md object-cover"
        />
      </AspectRatio>
    </div>
  ),
};

export const Portrait: Story = {
  render: () => (
    <div className="w-[300px]">
      <AspectRatio ratio={3 / 4} className="bg-muted">
        <img
          src="https://placehold.co/300x400"
          alt="Placeholder Alt Text"
          className="rounded-md object-cover"
        />
      </AspectRatio>
    </div>
  ),
};
