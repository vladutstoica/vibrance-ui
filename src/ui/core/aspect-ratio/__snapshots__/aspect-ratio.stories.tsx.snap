// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design System/Core/AspectRatio Landscape smoke-test 1`] = `
<div class="w-[500px]">
  <div data-radix-aspect-ratio-wrapper
       style="position: relative; width: 100%; padding-bottom: 56.25%;"
  >
    <div class="bg-muted"
         style="position: absolute; inset: 0px;"
    >
      <img alt="Placeholder Alt Text"
           class="rounded-md object-cover"
           src="https://placehold.co/1600x900"
      >
    </div>
  </div>
</div>
`;

exports[`Design System/Core/AspectRatio Portrait smoke-test 1`] = `
<div class="w-[300px]">
  <div data-radix-aspect-ratio-wrapper
       style="position: relative; width: 100%; padding-bottom: 133.333%;"
  >
    <div class="bg-muted"
         style="position: absolute; inset: 0px;"
    >
      <img alt="Placeholder Alt Text"
           class="rounded-md object-cover"
           src="https://placehold.co/300x400"
      >
    </div>
  </div>
</div>
`;

exports[`Design System/Core/AspectRatio Square smoke-test 1`] = `
<div class="w-[500px]">
  <div data-radix-aspect-ratio-wrapper
       style="position: relative; width: 100%; padding-bottom: 100%;"
  >
    <div class="bg-muted"
         style="position: absolute; inset: 0px;"
    >
      <img alt="Placeholder Alt Text"
           class="rounded-md object-cover"
           src="https://placehold.co/500"
      >
    </div>
  </div>
</div>
`;
